modifytags/modifytags.go

```diff
@@ -0,0 +1,372 @@
+		tags.AddOptions(key, val...)
+	}
+
+	return tags, nil
+}
+
+func (mod *Modification) addTags(fieldName string, tags *structtag.Tags) (*structtag.Tags, error) {
+	if len(mod.Add) == 0 {
+		return tags, nil
+	}
+
+	split := camelcase.Split(fieldName)
+	name := ""
+
+	unknown := false
+	switch mod.Transform {
+	case SnakeCase:
+		var lowerSplit []string
+		for _, s := range split {
+			s = strings.Trim(s, "_")
+			if s == "" {
+				continue
+			}
+			lowerSplit = append(lowerSplit, strings.ToLower(s))
+		}
+
+		name = strings.Join(lowerSplit, "_")
+	case LispCase:
+		var lowerSplit []string
+		for _, s := range split {
+			lowerSplit = append(lowerSplit, strings.ToLower(s))
+		}
+
+		name = strings.Join(lowerSplit, "-")
+	case CamelCase:
+		var titled []string
+		for _, s := range split {
+			titled = append(titled, strings.Title(s))
+		}
+
+		titled[0] = strings.ToLower(titled[0])
+
+		name = strings.Join(titled, "")
+	case PascalCase:
+		var titled []string
+		for _, s := range split {
+			titled = append(titled, strings.Title(s))
+		}
+
+		name = strings.Join(titled, "")
+	case TitleCase:
+		var titled []string
+		for _, s := range split {
+			titled = append(titled, strings.Title(s))
+		}
+
+		name = strings.Join(titled, " ")
+	case Keep:
+		name = fieldName
+	default:
+		unknown = true
+	}
+
+	if mod.ValueFormat != "" {
+		prevName := name
+		name = strings.ReplaceAll(mod.ValueFormat, "{field}", name)
+		if name == mod.ValueFormat {
+			// support old style for backward compatibility
+			name = strings.ReplaceAll(mod.ValueFormat, "$field", prevName)
+		}
+	}
+
+	for _, key := range mod.Add {
+		split = strings.SplitN(key, ":", 2)
+		if len(split) >= 2 {
+			key = split[0]
+			name = strings.Join(split[1:], "")
+		} else if unknown {
+			// the user didn't pass any value but want to use an unknown
+			// transform. We don't return above in the default as the user
+			// might pass a value
+			return nil, fmt.Errorf("unknown transform option %q", mod.Transform)
+		}
+
+		tag, err := tags.Get(key)
+		if err != nil {
+			// tag doesn't exist, create a new one
+			tag = &structtag.Tag{
+				Key:  key,
+				Name: name,
+			}
+		} else if mod.Overwrite {
+			tag.Name = name
+		}
+
+		if err := tags.Set(tag); err != nil {
+			return nil, err
+		}
+	}
+
+	return tags, nil
+}
+
+func isPublicName(name string) bool {
+	for _, c := range name {
+		return unicode.IsUpper(c)
+	}
+	return false
+}
+
+// validate determines whether the Modification is valid or not.
+func (mod *Modification) validate() error {
+	if len(mod.Add) == 0 &&
+		len(mod.AddOptions) == 0 &&
+		!mod.Clear &&
+		!mod.ClearOptions &&
+		len(mod.RemoveOptions) == 0 &&
+		len(mod.Remove) == 0 {
+		return errors.New("one of " +
+			"[-add-tags, -add-options, -remove-tags, -remove-options, -clear-tags, -clear-options]" +
+			" should be defined")
+	}
+	return nil
+}
+
+func quote(tag string) string {
+	return "`" + tag + "`"
+}
+
+type RewriteErrors struct {
```

Comment:
// RewriteErrors is an error describing errors that occurred while rewriting struct field tags.

(or similar).